src/pointer/ptr.rs

```diff
@@ -674,6 +687,38 @@ mod _casts {
         }
     }
 
+    impl<'a, T, I> Ptr<'a, T, I>
+    where
+        T: 'a,
+        I: Invariants<Validity = invariant::Initialized>,
+    {
+        /// Casts this pointer-to-initialized into a pointer-to-bytes.
+        #[allow(clippy::wrong_self_convention)]
+        pub(crate) fn as_bytes(
+            self,
+        ) -> Ptr<'a, [u8], (I::Aliasing, invariant::Aligned, invariant::Valid)> {
+            // SAFETY: We ensure that:
+            // - `cast(p)` is implemented exactly as follows: `|p: *mut T| p as
+            //   *mut U`.
```

Comment:
This isn't true anymore. Can you justify that this is equivalent, or update the safety precondition so that this is documented as valid?

Reply:
Some days, I truly feel like one of those people who writes `MaybeUninit::uninit().assume_init()` lol

Reply:
Done! `cast_unsized` is now generalized to specify that `cast(p)` can be any kind of cast, where a 'cast' is a provenance-preserving transformation of a pointer-like value into another pointer-like value to the same allocation. To minimize safety-comment churn at callsites, I provide callers with a choose-your-own-adventure of proof outlines.